test/e2e_node/separate_disk_test.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+)
+
+// Eviction Policy is described here:
+// https://github.com/kubernetes/design-proposals-archive/blob/main/node/kubelet-eviction.md
+// Existing Eviction tests are quite flaky and need some investigation
+// Flakiness arises from ordering
+// Stats is best effort and we evict based on stats being successful
+
+// Container runtime filesystem should display different stats for imagefs and nodefs
+var _ = SIGDescribe("Summary", feature.SeparateDiskTest, func() {
+	f := framework.NewDefaultFramework("summary-test")
+	f.It("should display different stats for imagefs and nodefs", func(ctx context.Context) {
+		summary := eventuallyGetSummary(ctx)
+		// Available and Capacity are the most useful to tell difference
+		gomega.Expect(summary.Node.Fs.AvailableBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.AvailableBytes))
+		gomega.Expect(summary.Node.Fs.CapacityBytes).ToNot(gomega.Equal(summary.Node.Runtime.ImageFs.CapacityBytes))
+
+	})
+})
+
+// Node disk pressure is induced by consuming all inodes on the Writeable Layer (imageFS).
+var _ = SIGDescribe("InodeEviction", framework.WithSlow(), framework.WithSerial(), framework.WithDisruptive(), feature.SeparateDiskTest, func() {
```

Comment:
There is a lot of code duplications in this file and `eviction_test.go`. Can we wrap the code in a helper functions and call them from both?